src/EntityFramework/Infrastructure/DbQuery.cs

```diff
@@ -217,7 +219,21 @@ public DbQuery<TElement> Cast<TElement>()
         /// <returns> The query string. </returns>
         public override string ToString()
         {
-            return InternalQuery == null ? base.ToString() : InternalQuery.ToString();
+            return InternalQuery == null ? base.ToString() : InternalQuery.ToTraceString();
+        }
+
+        private string DebuggerDisplay()
+        {
+            return base.ToString();
+        }
+
+        /// <summary>
+        /// Returns the SQL that will be run against the underlying data store when this query is executed.
+        /// </summary>
+        /// <returns> The SQL string. </returns>
+        public string Sql
```

Comment:
I don't feel strongly, but the name `Sql` seems correct and unambiguous enough to me.